### ISSUE TITLE
`--execute` doesn't work with a `julia` kernel

### ISSUE THREAD

**Issue by diegozea** [JuliaLang/IJulia.jl#410]
Works with Python 2 kernel:

```
diego@SIRRAH:~/Desktop$ sudo ipython nbconvert --to=html --template=basic --execute ej.ipynb 
[NbConvertApp] Converting notebook ej.ipynb to html
[NbConvertApp] Executing notebook with kernel: python2
[NbConvertApp] Writing 882 bytes to ej.html
```

But doesn't work with a Julia kernel:

```
diego@SIRRAH:~/Desktop$ sudo ipython nbconvert --to=html --template=basic --execute Untitled.ipynb 
[NbConvertApp] Converting notebook Untitled.ipynb to html
[NbConvertApp] Executing notebook with kernel: julia-0.4
Starting kernel event loops.
[NbConvertApp] ERROR | failed to run cell: KeyError('status',)
[NbConvertApp] ERROR | println("Hola mundo")
Traceback (most recent call last):
  File "/usr/local/bin/ipython", line 11, in <module>
    sys.exit(start_ipython())
  File "/home/diego/.local/lib/python2.7/site-packages/IPython/__init__.py", line 118, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/home/diego/.local/lib/python2.7/site-packages/traitlets/config/application.py", line 592, in launch_instance
    app.start()
  File "/home/diego/.local/lib/python2.7/site-packages/IPython/terminal/ipapp.py", line 349, in start
    return self.subapp.start()
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/nbconvertapp.py", line 286, in start
    self.convert_notebooks()
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/nbconvertapp.py", line 409, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/nbconvertapp.py", line 380, in convert_single_notebook
    output, resources = self.export_single_notebook(notebook_filename, resources)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/nbconvertapp.py", line 332, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/exporters/exporter.py", line 166, in from_filename
    return self.from_notebook_node(nbformat.read(f, as_version=4), resources=resources, **kw)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/exporters/html.py", line 65, in from_notebook_node
    return super(HTMLExporter, self).from_notebook_node(nb, resources, **kw)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/exporters/templateexporter.py", line 198, in from_notebook_node
    nb_copy, resources = super(TemplateExporter, self).from_notebook_node(nb, resources, **kw)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/exporters/exporter.py", line 131, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/exporters/exporter.py", line 303, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/preprocessors/base.py", line 47, in __call__
    return self.preprocess(nb,resources)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/preprocessors/execute.py", line 83, in preprocess
    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/preprocessors/base.py", line 70, in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/preprocessors/execute.py", line 97, in preprocess_cell
    outputs = self.run_cell(cell)
  File "/home/diego/.local/lib/python2.7/site-packages/nbconvert/preprocessors/execute.py", line 128, in run_cell
    if msg['metadata']['status'] == 'error' and not self.allow_errors:
KeyError: 'status'

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@scipy.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    c.Application.verbose_crash=True

```

**Comment by minrk** [comment]
I believe this is fixed in master, which should become 4.2 before long. Can you test with master?

**Comment by stevengj** [comment]
Closing as no longer reproducible.
